src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -811,6 +813,18 @@ DeferredErrorIfUnsupportedRecurringTuplesJoin(
 			 */
 			if (RelationInfoContainsOnlyRecurringTuples(plannerInfo, outerrelRelids))
 			{
+				/*
+				 * These constraints will be validated in worker nodes, so no
+				 * need to run them on coordinator. In fact, we disable their
+				 * execution: check CitusExecutorRun. We don't care whether
+				 * we can pushdown the subquery or not. Hence we simply
+				 * consider this as a supported Recurring Tuples Join.
+				 */
+				if (SimpleAlterTableConstraintCheck(plannerInfo->parse))
+				{
+					continue;
```

Comment:
I feel a bit uncomfortable with generating an incorrect query plan and still doing some amount of work. Is there a way to skip the work in CitusExecutorRun?

Reply:
This work is already skipped in CitusExecutorRun.

So the error happens because the partition is not distributed but its a local PG table. If we run the command with an already distributed partition, it doesn't error out in planning phase and all the work is done for it => it is considered as proper pushdown query. But then it is skipped in CitusExecutorRun anyway.

Reply:
Got it, so we mainly want to avoid the planner throwing an error. Could we generate some dummy plan early on in the planner?

Reply:
Or, could we treat it as a not-distributed query?

Reply:
Thanks for your input. If I can figure out a way early on in the planner to not consider it as a distributed query that would make more sense as it aligns with skipping the query in CitusExecutorRun.